### PR TITLE
Log failed validation in GlobalL0Service.pullLatestSnapshot

### PR DESCRIPTION
**Pull request by sdavidp**

### CODE REVIEW

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala [hunk 1]
```diff
@@ -111,6 +111,10 @@ object GlobalL0Service {
         type Result = LatestSnapshotTuple
         for {
           peers <- globalL0ClusterStorage.getPeers.map(nes => Random.shuffle(nes.toSortedSet))
+
+          _ <- logger.debug(s"Pulling latest snapshot using ${peers.size} peers")
+          _ <- logger.trace(s"${peers.show}")
```
Comment (TheMMaciek) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala:116]: Even if it's a trace log I would still log a short peer id format if it's not the case right now.

Comment (TheMMaciek) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala:116]: There is a very little chance that there will be a clash on e.g. first 5 chars for Id. Hence logging more than that seems redundant.

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala [hunk 2]
```diff
@@ -137,12 +141,33 @@ object GlobalL0Service {
...
+          case Some(ord) =>
+            logger.debug(s"Majority ordinal ${ord.show} minus Snapshot ordinal ${snapshot.ordinal.show} not in $ordinalRange").as(false)
+        }
+
+      private def majorityHashValidation(snapshot: Hashed[GlobalIncrementalSnapshot], majorityPeers: NonEmptyList[L0Peer]): F[Boolean] =
+        getMajorityHash(majorityPeers, snapshot.ordinal).flatMap {
+          case None                                 => logger.debug(s"No majority hash for ordinal ${snapshot.ordinal.show}").as(false)
+          case Some(hash) if hash === snapshot.hash => true.pure[F]
+          case Some(hash) => logger.debug(s"Majority/Snapshot hash mismatch: ${hash.show} != ${snapshot.hash.show}").as(false)
+        }
```
[the remark this answers is not in the document]
Comment (sdavidp) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala:169]: @TheMMaciek I agree about the code being cluttered but it wasn't clear why validation failed. Can we talk about this further in a Huddle?

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala [hunk 3]
```diff
@@ -137,12 +142,36 @@ object GlobalL0Service {
...
+          .map(_.isValid)
+
+      private def majorityOrdinalValidation(snapshot: Hashed[GlobalIncrementalSnapshot], majorityPeers: NonEmptyList[L0Peer]): F[Boolean] =
+        getMajorityOrdinal(majorityPeers)
+          .map(_.map(ord => (ord, ordinalRange.contains(ord.value - snapshot.ordinal.value))))
+          .flatMap {
+            case Some((_, true)) => true.pure[F]
+            case maybe =>
+              logger.debug(s"Majority ordinal ${maybe.map(_._1.show)} missing or not in $ordinalRange, ${snapshot.ordinal.show}").as(false)
+          }
```
Comment (TheMMaciek) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala:164]: My proposal:
```
      private def majorityOrdinalValidation(snapshot: Hashed[GlobalIncrementalSnapshot], majorityPeers: NonEmptyList[L0Peer]): F[Boolean] =
        getMajorityOrdinal(majorityPeers).flatMap { maybeMajorityOrdinal =>
          val isInRange = maybeMajorityOrdinal.exists(o => ordinalRange.contains(o.value - snapshot.ordinal.value))

          logger
            .debug(s"Majority ordinal ${maybeMajorityOrdinal.show} missing or not in $ordinalRange, ${snapshot.ordinal.show}")
            .unlessA(isInRange)
            .as(isInRange)
        }
```

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala [hunk 4]
```diff
@@ -137,12 +142,36 @@ object GlobalL0Service {
...
+
+      private def majorityHashValidation(snapshot: Hashed[GlobalIncrementalSnapshot], majorityPeers: NonEmptyList[L0Peer]): F[Boolean] =
+        getMajorityHash(majorityPeers, snapshot.ordinal)
+          .map(_.map(h => (h, h === snapshot.hash)))
+          .flatMap {
+            case Some((_, true)) => true.pure[F]
+            case maybe =>
+              logger.debug(s"Majority/Snapshot hash mismatch: ${maybe.map(_._1.show)}, ${snapshot.hash.show}").as(false)
+          }
```
Comment (TheMMaciek) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala:173]: My proposal:
```
      private def majorityHashValidation(snapshot: Hashed[GlobalIncrementalSnapshot], majorityPeers: NonEmptyList[L0Peer]): F[Boolean] =
        getMajorityHash(majorityPeers, snapshot.ordinal).flatMap { maybeMajorityHash =>
          val isMajority = maybeMajorityHash.exists(_ === snapshot.hash)

          logger
            .debug(s"Majority/Snapshot hash mismatch: ${maybeMajorityHash.show}, ${snapshot.hash.show}")
            .unlessA(isMajority)
            .as(isMajority)
        }
```

**Review by TheMMaciek — APPROVED** [review]

**Review by TheMMaciek — APPROVED** [review]

**Pull request merged**